tests/core/tests/test_resources/test_import_export.py

```diff
@@ -81,6 +82,35 @@ class Meta:
             fields = ["id", "price", "name"]
             model = Book
 
+    class DeclaredModelFieldBookResource(BaseBookResource):
+        # Non-model field, should come after model fields by default
+        author_full_name = fields.Field(
+            attribute="author",
+            column_name="author full name",
+        )
+
+        # Order of declared fields in `ModelResource` shouldn't change export order
+        categories = fields.Field(
+            attribute="categories",
+            column_name="categories",
+            widget=widgets.ManyToManyWidget(model=Category, field="name"),
+        )
+        published = fields.Field(
+            attribute="published",
+            column_name="published",
+            widget=widgets.DateWidget("%d.%m.%Y"),
+        )
+        author = fields.Field(attribute="author__name", column_name="author")
+
+        class Meta:
+            model = Book
+
+        def dehydrate_author_full_name(self, obj):
```

Comment:
Do we need this method in the test?  Presumably it can be removed (and declaration on line 259)

Reply:
Originally I wasn't even going to create another resource and just inherit `DeclaredModelFieldBookResource` however I have noticed a possible bug where resource inheritance is not working properly when it is inherited more than twice.

The reason for dehydration method is testing all possible outcomes for declaring a field. The test was passing without that part but just to cover all possibilities.

Reply:
OK thanks - happy to leave the dehydrate method in.

> I have noticed a possible bug where resource inheritance is not working properly when it is inherited more than twice.

Probably [this one](https://github.com/django-import-export/django-import-export/issues/1363).